plugin/src/main/java/org/gradle/testretry/internal/RetryTestResultProcessor.java

```diff
@@ -68,11 +68,20 @@ public void started(TestDescriptorInternal descriptor, TestStartEvent testStartE
 
     @Override
     public void completed(Object testId, TestCompleteEvent testCompleteEvent) {
-        activeDescriptorsById.remove(testId);
+
+        final TestDescriptorInternal descriptor = activeDescriptorsById.get(testId);
```

Comment:
nit: we don't seem to be consistent in our use of `final` in this class. We should either always use it when possible, or never.

Reply:
```suggestion
        final TestDescriptorInternal descriptor = activeDescriptorsById.remove(testId);
```